### ISSUE TITLE
Issue when trying to acquire GKE Cluster

### ISSUE THREAD

**Issue by travisrandolph-bestbuy** [GoogleCloudPlatform/k8s-config-connector#568]
## Bug Description

When trying to acquire a GKE cluster that has already been created I'm seeing the error below. I'm not specifying a node version, so I'm not sure why it's complaining about it.

```bash
Name:         {{REDACTED}}
Annotations:  cnrm.cloud.google.com/management-conflict-prevention-policy: none
              cnrm.cloud.google.com/project-id: {{REDACTED}}
              cnrm.cloud.google.com/remove-default-node-pool: true
              cnrm.cloud.google.com/state-into-spec: merge
API Version:  container.cnrm.cloud.google.com/v1beta1
Kind:         ContainerCluster
Metadata:
Spec:
  Addons Config:
    Http Load Balancing:
      Disabled:  false
  Database Encryption:
    Key Name:                   {{REDACTED}}
    State:                      ENCRYPTED
  Enable Binary Authorization:  true
  Enable Shielded Nodes:        true
  Initial Node Count:           1
  Ip Allocation Policy:
    clusterIpv4CidrBlock:          {{REDACTED}}
    Cluster Secondary Range Name:   {{REDACTED}}
    servicesIpv4CidrBlock:          {{REDACTED}}
    Services Secondary Range Name:  {{REDACTED}}
  Location:                         us-central1
  Master Authorized Networks Config:
    Cidr Blocks:
      Cidr Block:  {{REDACTED}}
  Network Ref:
    Name:  {{REDACTED}}
  Node Config:
    Image Type:    COS
    Machine Type:  n1-standard-8
    Metadata:
      Block - Project - Ssh - Keys:  true
      Disable - Legacy - Endpoints:  true
    Oauth Scopes:
      https://www.googleapis.com/auth/cloud-platform
    Shielded Instance Config:
      Enable Integrity Monitoring:  true
      Enable Secure Boot:           true
    Tags:
      {{REDACTED}}
  Node Locations:
    us-central1-b
    us-central1-c
  Notification Config:
    Pubsub:
      Enabled:  true
      Topic Ref:
        Name:  gke-notification
  Private Cluster Config:
    Enable Private Endpoint:  false
    Enable Private Nodes:     true
    masterIpv4CidrBlock:      {{REDACTED}}
  Release Channel:
    Channel:  stable
  Subnetwork Ref:
    Name:  {{REDACTED}}
  Vertical Pod Autoscaling:
    Enabled:  true
  Workload Identity Config:
    Identity Namespace:  {{REDACTED}}
Status:
  Conditions:
    Last Transition Time:  2021-11-01T15:10:03Z
    Message:               Update call failed: error calculating diff: 1 error occurred:
                           * node_version can only be specified if remove_default_node_pool is not true


    Reason:             UpdateFailed
    Status:             False
    Type:               Ready
  Observed Generation:  1
Events:
  Type     Reason        Age               From                         Message
  ----     ------        ----              ----                         -------
  Warning  UpdateFailed  8s (x3 over 11s)  containercluster-controller  Update call failed: error calculating diff: 1 error occurred:
           * node_version can only be specified if remove_default_node_pool is not true
```

### Kubernetes Cluster Version

Client Version: v1.19.2
Server Version: v1.20.10-gke.301

### Config Connector Version

1.60.0

### Config Connector Mode

namespaced

### YAML
```yaml
apiVersion: container.cnrm.cloud.google.com/v1beta1
kind: ContainerCluster
metadata:
  annotations:
    cnrm.cloud.google.com/remove-default-node-pool: "true"
  name: {{REDACTED}}
spec:
  location: us-central1
  nodeLocations:
  - us-central1-b
  - us-central1-c
  initialNodeCount: 1
  releaseChannel:
      channel: stable
  notificationConfig:
    pubsub:
      enabled: true
      topicRef:
        name: gke-notification
  enableBinaryAuthorization: true
  verticalPodAutoscaling:
    enabled: true
  networkRef:
    name: {{REDACTED}}
  subnetworkRef:
    name: {{REDACTED}}
  addonsConfig:
    httpLoadBalancing:
      disabled: false
  masterAuthorizedNetworksConfig:
    cidrBlocks:
    - cidrBlock: {{REDACTED}}
  privateClusterConfig:
    enablePrivateEndpoint: false
    enablePrivateNodes: true
    masterIpv4CidrBlock: {{REDACTED}}
  ipAllocationPolicy:
    clusterIpv4CidrBlock: {{REDACTED}}
    clusterSecondaryRangeName: gke-pods
    servicesIpv4CidrBlock: {{REDACTED}}
    servicesSecondaryRangeName: gke-services
  databaseEncryption:
    state: ENCRYPTED
    keyName: {{REDACTED}}
  enableShieldedNodes: true
  workloadIdentityConfig:
    identityNamespace: {{REDACTED}}
  nodeConfig:
    imageType: COS
    metadata:
      disable-legacy-endpoints: "true"
      block-project-ssh-keys: "true"
    oauthScopes:
    - https://www.googleapis.com/auth/cloud-platform
    shieldedInstanceConfig:
      enableSecureBoot: true
      enableIntegrityMonitoring: true
    machineType: n1-standard-8
    tags:
    - {{REDACTED}}
```

**Comment by jcanseco** [comment]
Hi @travisrandolph-bestbuy -- looking. Can you share a copy of the YAML you used to acquire the cluster?

**Comment by travisrandolph-bestbuy** [comment]
@jcanseco I added the yaml above.

**Comment by travisrandolph-bestbuy** [comment]
I think I see the issue now. Even thought I want to kill the default node pool, it's still trying to add the field to my resource. Is there a way to have KCC ignore the default node pool config if we have the annotation set to delete the default node pool?

**Comment by jcanseco** [comment]
I see, interesting, you might be right. I am not able to reproduce the issue yet, so I can't make a confident and specific recommendation just yet, but are you able to unblock yourself in the meantime by deleting the default node pool and acquiring the cluster in separate steps?

**Comment by travisrandolph-bestbuy** [comment]
I just had to match up the config for the default node pool that was being deleted. Our current pattern deploys a GPU node pool so we had to have a larger machine type for the default node pool in case an engineer deleted the cluster from the console. In that case KCC will deploy the cluster again.

**Comment by jcanseco** [comment]
@travisrandolph-bestbuy I apologize for the late reply. I have been attempting to reproduce the issue (the issue of failing to acquire an existing cluster), but I have had no luck so far.

Also, according to the code, the following error should only ever show up when you're creating a new cluster, so I am not sure why it would occur during an acquisition:
```
node_version can only be specified if remove_default_node_pool is not true
```

My question would be: Are you still consistently facing this issue? If so, it would help us to see the state of the cluster on GCP (`gcloud container clusters describe`) in addition to the YAML you used to try to acquire it, so we can better see what the diffs were.

Lastly, I'm not sure if the following is contributing anythign to the problem, but I believe you shouldn't be setting `nodeConfig` in your YAML if your intention is to delete the default node pool anyway, since:
1. `nodeConfig` represents the config of the default node pool
2. GKE actually updates `nodeConfig` to point to your "first" custom node pool once the default pool is deleted, so you want to let `nodeConfig` float, otherwise KCC will error out trying to update the underlying `nodeConfig`.

**Comment by travisrandolph-bestbuy** [comment]
Hello @jcanseco. I actually moved on from this. I removed the default node pool config from our chart. The reason I ever went down this path is because we were having issues after someone deleted a cluster from the console manually. When KCC was trying to add it back it was complaining that the default node pool machine type doesn't support GPU. This is because we had a GPU pool added as well. I just went ahead and fixed the config for that specific cluster.
